commit ce13987ad8ecb65e414573d0ee2ac0062c5135a6
Author: Michael Still <mikal@stillhq.com>
Date:   Tue Jan 20 15:25:44 2026 +1100

    Allow setting of the new SPICE debug_logging option.
    
    This emits debug logs to the libvirt qemu log if enabled, which
    can be helpful when debugging connnectivity issues.
    
    Change-Id: Ia3686538952ec4b88877c2ae4f5c1699416ec851
    Signed-off-by: Michael Still <mikal@stillhq.com>

diff --git a/ansible/group_vars/all/nova.yml b/ansible/group_vars/all/nova.yml
index 24cfb9914..845ed12e8 100644
--- a/ansible/group_vars/all/nova.yml
+++ b/ansible/group_vars/all/nova.yml
@@ -76,3 +76,4 @@ nova_spice_playback_compression: true
 nova_spice_streaming_mode: "filter"
 nova_spice_require_secure: false
 nova_spice_allow_concurrent: true
+nova_spice_debug_logging: "no"
diff --git a/ansible/roles/nova-cell/templates/nova.conf.j2 b/ansible/roles/nova-cell/templates/nova.conf.j2
index c32c3c660..8da5e6b66 100644
--- a/ansible/roles/nova-cell/templates/nova.conf.j2
+++ b/ansible/roles/nova-cell/templates/nova.conf.j2
@@ -63,6 +63,7 @@ enabled = false
 enabled = true
 require_secure = {{ nova_spice_require_secure }}
 allow_concurrent = {{ nova_spice_allow_concurrent | bool | lower }}
+debug_logging = {{ nova_spice_debug_logging | bool | lower }}
 server_listen = {{ api_interface_address }}
 image_compression = {{ nova_spice_image_compression }}
 jpeg_compression = {{ nova_spice_jpeg_compression }}
diff --git a/doc/source/reference/compute/nova-guide.rst b/doc/source/reference/compute/nova-guide.rst
index a491cb56a..4c7ba310e 100644
--- a/doc/source/reference/compute/nova-guide.rst
+++ b/doc/source/reference/compute/nova-guide.rst
@@ -69,6 +69,8 @@ through to ``nova``:
   playback. Defaults to ``yes``.
 * nova_spice_streaming_mode: what streaming mode to use with SPICE consoles.
   Defaults to ``filter``.
+* nova_spice_debug_logging: whether to emit debug logs from SPICE sessions to
+  the libvirt qemu log. Defaults to ``no``.
 
 Cells
 =====
